packages/cli/src/commands/inspect.ts

```diff
@@ -98,12 +135,22 @@ export default async function main(client: Client) {
     `Fetching deployment "${deploymentIdOrHost}" in ${chalk.bold(contextName)}`
   );
 
+  const until = Date.now() + timeout;
+  const wait = argv['--wait'];
+
   // resolve the deployment, since we might have been given an alias
-  const deployment = await getDeployment(
-    client,
-    contextName,
-    deploymentIdOrHost
-  );
+  let deployment = await getDeployment(client, contextName, deploymentIdOrHost);
+
+  while (Date.now() < until) {
+    if (!wait || !isDeploying(deployment.readyState)) {
+      break;
+    }
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏check `wait` option first

The logic here does what it's supposed to do, but it was confusing to read at first. I was worried that this was always waiting, but a careful readthrough and testing showed that's not true.

I think this would make more sense if it started with a check on the flag that enables this entire behavior, like:

```
if (wait) {
  // ...
}
```